File: backend/databasehandler/src/main/java/com/pixtranslator/backend/databasehandler/model/EnglishWord.java
package com.pixtranslator.backend.databasehandler.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;

@Entity
public class EnglishWord implements Serializable {
  @Id
  @Getter
  @Setter
  @Column(unique = true)
  private String word;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "germanword_englishword")
  @JsonBackReference
  @Getter
  @Setter
  private GermanWord germanword;
}

File: backend/databasehandler/src/main/kotlin/com/pixtranslator/backend/databasehandler/model/EnglishWord.kt
package com.pixtranslator.backend.databasehandler.model

import com.fasterxml.jackson.annotation.JsonBackReference
import java.io.Serializable
import javax.persistence.*

@Entity
class EnglishWord : Serializable {
  @Id
  @Column(unique = true)
  var word: String? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "germanword_englishword")
  @JsonBackReference
  var germanword: GermanWord? = null
}
